app.rb

```diff
@@ -16,15 +59,280 @@ class MyApp < Sinatra::Base
     end
   end
 
-  error do |e|
-    if e.is_a? ActiveRecord::RecordNotFound
-      halt 404, json(error: "Not Found")
-    elsif e.is_a? ActiveRecord::RecordInvalid
-      halt 422, json(error: e.message)
+
+  def require_login!
+    unless current_user
+    redirect '/'
+    end
+  end
+
+  before do
+    unless ["/", "/newuser", "/dashboard"].include?(request.path)
+      require_login!
+      return
+    end
+  end
+
+  if ENV['PRY_ON_ERROR']
+    error do |e|
+      binding.pry
+    end
+  end
+
+  # login page show
+  get '/' do
+    erb :login
+  end
+
+
+#account page
+  get '/account' do
+    erb :account
+  end
+
+  #reset password
+  get '/password' do
+    erb :password
+  end
+
+  post '/password' do
+    if (current_user.password == Digest::SHA256.hexdigest(params[:oldpassword])) && (params[:newpassword1] == params[:newpassword2])
+      change_password(params[:newpassword1])
```

Comment:
I very much like having this helper function. Would it be easier to reuse if it were actually a method on the `current_user / User` object instead?